cmd/clusterctl/client/cluster/mover_test.go

```diff
@@ -788,3 +788,86 @@ func Test_objectsMoverService_checkTargetProviders(t *testing.T) {
 		})
 	}
 }
+
+func Test_objectMoverService_ensureNamespace(t *testing.T) {
+	type fields struct {
+		fromProxy Proxy
+	}
+	type args struct {
+		toProxy   Proxy
+		namespace string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "ensureNamespace doesn't fail given an existing namespace",
+			// Create a fake local cluster in namespace-1
+			fields: fields{getObjectGraphWithObjs(test.NewFakeCluster("namespace-1", "cl1").Objs()).proxy},
+			args: args{
+				// Create a fake cluster target with namespace-1 already existing
+				toProxy: getFakeProxyWithCRDs().WithObjs(test.NewFakeCluster("namespace-1", "cl1").Objs()...),
```

Comment:
This can be simplified using `toProxy: test.NewFakeProxy().WithObjs(namespace1)` where `namespace1` should be a namespace object you define in beforehand 
```
namespace1 := &corev1.Namespace{
			TypeMeta: metav1.TypeMeta{
				Kind: "Namespace",
			},
			ObjectMeta: metav1.ObjectMeta{
				Name:   "namespace-1",
				Labels: labels,
			},
		}
```